examples/platforms/efr32mg12/radio.c

```diff
@@ -186,7 +185,28 @@ static const RAIL_IEEE802154_Config_t sRailIeee802154Config = {
     .isPanCoordinator = false,
 };
 
-RAIL_DECLARE_TX_POWER_VBAT_CURVES(piecewiseSegments, curvesSg, curves24Hp, curves24Lp);
+static const uint8_t piecewiseSegments = RAIL_PA_CURVES_2P4_HP_SG_PIECEWISE_SEGMENTS;
+
+#if RADIO_CONFIG_2P4GHZ_PA_USES_DCDC
+RAIL_DECLARE_PA_2P4_HP_DCDC;
+static const RAIL_TxPowerCurves_t curves24Hp[1] = {
+    {RAIL_PA_CURVES_2P4_HP_DCDC_MAX_POWER, RAIL_PA_CURVES_2P4_HP_DCDC_MIN_POWER, twoPointFourHpDcdcCurves}};
+#else
+RAIL_DECLARE_PA_2P4_HP_VBAT;
+static const RAIL_TxPowerCurves_t curves24Hp[1] = {
+    {RAIL_PA_CURVES_2P4_HP_VBAT_MAX_POWER, RAIL_PA_CURVES_2P4_HP_VBAT_MIN_POWER, twoPointFourHpVbatCurves}};
+#endif
+static const int16_t curves24Lp[RAIL_PA_CURVES_LP_VALUES] = RAIL_PA_CURVES_2P4_LP;
+
+#if RADIO_CONFIG_SUBGIG_PA_USES_DCDC
+RAIL_DECLARE_PA_SG_DCDC;
+static const RAIL_TxPowerCurves_t curvesSg[1] = {
+    {RAIL_PA_CURVES_SG_DCDC_MAX_POWER, RAIL_PA_CURVES_SG_DCDC_MIN_POWER, subgigDcdcCurves}};
+#else
+RAIL_DECLARE_PA_SG_VBAT;
+static const RAIL_TxPowerCurves_t curvesSg[1] = {
+    {RAIL_PA_CURVES_SG_VBAT_MAX_POWER, RAIL_PA_CURVES_SG_VBAT_MIN_POWER, subgigVbatCurves}};
+#endif
```

Comment:
It would be ideal if these declarations could be provided by the RAIL header. In fact, the header provides `RAIL_DECLARE_TX_POWER_VBAT_CURVES` and `RAIL_DECLARE_TX_POWER_DCDC_CURVES`, but those macros assume that both PAs are from by the same power rail.